mkcodes.py

```diff
@@ -13,48 +13,79 @@
     from markdown.extensions import Extension
     from markdown.treeprocessors import Treeprocessor
 
-
-def github_codeblocks(filepath, safe):
-    codeblocks = []
+# much easier to write the other names that an extension is known by
+ext_map = {
```

Comment:
Is there a specification or documentation of these language/extension mappings?

Reply:
Nope. this is just my freehand mapping of versions that I've seen in syntax highlighters to the language they output. 
Given the wide variety of syntax highlighters out there I'd expect this to evolve. 
Any ideas about where to make this mapping more formal is definitely welcome.

Reply:
for context: https://github.com/Python-Markdown/markdown/blob/master/markdown/extensions/fenced_code.py
python markdown just takes whatever you write as the language and passes it off to the pygments highliter. Which works great for java and py, but doesn't work well with csharp. Rather than cover everything, I just wanted to make it really clear where to add extensions and the mappings for other languages.

Reply:
For lack of a better option, I'm tempted to incorporate pygments' mappings. https://github.com/pygments/pygments/blob/master/pygments/lexers/_mapping.py

Reply:
Urgh - the file extensions are kind of secondary there.
I took a stab at this by chopping it up in vim: https://gist.github.com/mattkatz/8830cbf4e91d83e4ef29a5100bf9f601
It's LONG

Reply:
Just seems like a YAGNI moment - if there is increased appetite for lots of languages, that's a good time to refactor out lots of languages into a file with extension mappings - but if not, it's hard to check for acuracy or test or port over changes from pygments.

Reply:
Fair enough. I'd probably want to script the generation of that ext_map.py file so it can be easily updated.